letsencrypt/client/reverter.py

```diff
@@ -0,0 +1,434 @@
+                self._recover_checkpoint(self.direc['temp'])
+            except errors.LetsEncryptReverterError:
+                # We have a partial or incomplete recovery
+                logging.fatal("Incomplete or failed recovery for %s",
+                              self.direc['temp'])
+                raise errors.LetsEncryptReverterError(
+                    "Unable to revert temporary config")
+
+    def rollback_checkpoints(self, rollback=1):
+        """Revert 'rollback' number of configuration checkpoints.
+
+        :param int rollback: Number of checkpoints to reverse. A str num will be
+           cast to an integer. So '2' is also acceptable.
+
+        :raises :class:`letsencrypt.client.errors.LetsEncryptReverterError`: If
+            there is a problem with the input or if the function is unable to
+            correctly revert the configuration checkpoints.
+
+        """
+        try:
+            rollback = int(rollback)
+        except ValueError:
+            logging.error("Rollback argument must be a positive integer")
+            raise errors.LetsEncryptReverterError("Invalid Input")
+        # Sanity check input
+        if rollback < 0:
+            logging.error("Rollback argument must be a positive integer")
+            raise errors.LetsEncryptReverterError("Invalid Input")
+
+        backups = os.listdir(self.direc['backup'])
+        backups.sort()
+
+        if len(backups) < rollback:
+            logging.warning("Unable to rollback %d checkpoints, only %d exist",
+                            rollback, len(backups))
+
+        while rollback > 0 and backups:
+            cp_dir = os.path.join(self.direc['backup'], backups.pop())
+            try:
+                self._recover_checkpoint(cp_dir)
+            except errors.LetsEncryptReverterError:
+                logging.fatal("Failed to load checkpoint during rollback")
+                raise errors.LetsEncryptReverterError(
+                    "Unable to load checkpoint during rollback")
+            rollback -= 1
+
+    def view_config_changes(self):
+        """Displays all saved checkpoints.
+
+        All checkpoints are printed to the console.
+
+        .. todo:: Decide on a policy for error handling, OSError IOError...
+
+        """
+        backups = os.listdir(self.direc['backup'])
+        backups.sort(reverse=True)
+
+        if not backups:
+            logging.info("The Let's Encrypt client has not saved any backups "
+                         "of your configuration")
+            return
+        # Make sure there isn't anything unexpected in the backup folder
+        # There should only be timestamped (float) directories
+        try:
+            for bkup in backups:
+                float(bkup)
+        except ValueError:
+            raise errors.LetsEncryptReverterError(
+                "Invalid directories in {0}".format(self.direc['backup']))
+
+        for bkup in backups:
+            print time.ctime(float(bkup))
```

Comment:
I agree that this isn't good long-term.  At one point I did refactor them to be logging.info statements... but they don't appear on the screen too well.  I figured we needed a display method that wrapped the output and allowed you to scroll.

Perhaps the display.generic_notification window does do this...  I should have put a TODO... but I kind of wanted to look at/document/rearrange the display code before I spent time modifying this code.

I will do some quick experiments with generic_notification... might as well fix this now.